src/lib/transition_frontier/full_frontier/full_frontier.ml

```diff
@@ -960,12 +960,30 @@ module For_tests = struct
     let consensus_constants = precomputed_values.consensus_constants
   end
 
+  module T = Transaction_snark.Make (struct
```

Comment:
Although it is located in the `For_unit_tests` module, I suspect it can be initialized even by the production code if it's somehow referenced by some function somewhere.

I'm not sure if making the module is an expensive operation, but I'd suggest:

1. Making it singleton for usage in unit tests (this would also reduce the code duplication)
2. Measure the time to initialize it, and if it's non-negligble, surround with some laziness

Reply:
Previously initialization would happen from within a separate process, only when it's launched. Now it will happen if the module that contains this is somehow referenced by some code that is invoked in initialization, and it's possible that this is executed every time.

Once again, I'm not sure if making a functor is expensive itself or it is cheap, just generating a bunch of lazy values.